### PR TITLE
Feat/moving js to ts

### PR DESCRIPTION
**Pull request by andy-liubinski-stefanini**
It's easy to write code. It's hard to refactor code.

### CODE REVIEW

File: src/store/AppContext.tsx [hunk 1]
```diff
@@ -1,16 +1,33 @@
...
 
-function GlobalProvider({ children }) {
+
+// TODO after functions
+interface AppContextProps {
+  handleSearchButton: () => void;
+  searchFieldVisible: boolean;
+  setSearchFieldVisible: (visible: boolean) => void;
+  handleGeolocate: () => void;
+  weatherData: any; // ???
```
Comment (nkv843) [inline src/store/AppContext.tsx]: from api docs

File: src/store/AppContext.tsx [hunk 2]
```diff
@@ -1,16 +1,33 @@
...
+  formattedDates: any; // ???
```
Comment (nkv843) [inline src/store/AppContext.tsx]: you define this dates

File: src/store/AppContext.tsx [hunk 3]
```diff
@@ -1,16 +1,33 @@
...
+
+// TODO after functions
+interface AppContextProps {
+  handleSearchButton: () => void;
+  searchFieldVisible: boolean;
+  setSearchFieldVisible: (visible: boolean) => void;
+  handleGeolocate: () => void;
+  weatherData: any; // ???
+  formattedDates: any; // ??? 
+  handleUnitToggle: () => void;
```
Comment (nkv843) [inline src/store/AppContext.tsx:11]: no args?

File: src/store/AppContext.tsx [hunk 4]
```diff
@@ -1,16 +1,33 @@
...
+  isCelsius: boolean;
+  setSelectedLocation: (any) => void;  // ???
```
Comment (nkv843) [inline src/store/AppContext.tsx]: {
latitude: number, 
longitude: number,
}

**Review by nkv843 — COMMENTED** [review]
pls call me if you facing any problems

File: src/App.tsx [hunk 5]
```diff
@@ -1,8 +1,9 @@
 import { CitySection, InfoBox, ForecastBox } from './features/WeatherDisplay';
+import './index.scss'; // Import the SCSS file
```
Comment (nkv843) [inline src/App.tsx]: remove this comment :)

File: src/components/ForecastDay.tsx [hunk 6]
```diff
@@ -24,17 +23,17 @@ export const ForecastDay = ({ day, formattedDate }) => {
...
-        <img className="w-16 h-16" src={iconLink} alt="Weather icon" />
+      <div className="day-forecast--icon icon">
+        <img className="icon--img" src={iconLink} alt="Weather icon" />
       </div>
-      <div className="day-forecast--description text-sm">
+      <div className="day-forecast--description">
         {day ? `${day.description}` : `Loading...`}
       </div>
-      <div className="day-forecast--temps pb-4">
+      <div className="day-forecast--temps">
```
Comment (nkv843) [inline src/components/ForecastDay.tsx]: pls merge prev PR or rebase this branch

Comment (nkv843) [reply src/components/ForecastDay.tsx]: in order to keep css changes outside of this PR

File: src/components/InformationItem.tsx [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+import './styles.scss';
+
+export const InformationItem = ({ legend, info }) => {
+  return (
+    <div className="informations-box--item info-item">
+      <h3 className="info-item--title">{legend}</h3>
+      <p className="info-item--score">{info ?? `N/A`}</p>
+    </div>
+  );
+};
```
Comment (nkv843) [inline src/components/InformationItem.tsx]: add props interface

File: src/components/SearchHistoryItem.tsx [hunk 8]
```diff
@@ -0,0 +1,28 @@
+/* eslint-disable react/prop-types */
+import './styles.scss';
+
+import { CiCircleRemove } from 'react-icons/ci';
+
+export const SearchHistoryItem = ({
+  handleHistoryItem,
+  handleDeleteHistoryItem,
+  place,
+  i,
```
Comment (nkv843) [inline src/components/SearchHistoryItem.tsx]: i?

File: src/components/index.ts [hunk 9]
```diff
@@ -0,0 +1,3 @@
+export { ForecastDay } from './ForecastDay.tsx';
```
Comment (nkv843) [inline src/components/index.ts:1]: remove .tsx

File: src/features/NavBar/Search.tsx [hunk 10]
```diff
@@ -2,6 +2,7 @@ import { CiSearch } from 'react-icons/ci';
 import { AppContext } from '../../store';
 import { SearchHistory } from '.';
 import { useEffect, useContext, useRef, useState } from 'react';
+import './styles.scss'; // Import the SCSS file
```
Comment (nkv843) [inline src/features/NavBar/Search.tsx]: remove this comment

**Review by nkv843 — COMMENTED** [review]
please rebase this branch to hide css changes. 
also this should be draft PR since it's not ready to be deployed

File: src/components/ForecastDay.tsx [hunk 11]
```diff
@@ -1,7 +1,24 @@
...
+  icon: string;
+}
+
+interface FormattedDate {
+  formattedDate: string;
+}
+
+interface ForecastDayProps {
+  day: Day | null;
+  formattedDate: FormattedDate;
```
Comment (nkv843) [inline src/components/ForecastDay.tsx:17]: is it possible to make it flat? 

like {
day: Day | null;
formattedDate: string;
}

Comment (nkv843) [reply src/components/ForecastDay.tsx:17]: or even ForecastDayProps {
  maxTempC: number;
  minTempC: number;
  description: string;
  icon: string;
  formattedDate: string
}

File: src/components/SearchHistoryItem.tsx [hunk 12]
```diff
@@ -1,9 +1,14 @@
...
 import './styles.scss';
-
 import { CiCircleRemove } from 'react-icons/ci';
 
-export const SearchHistoryItem = ({
+interface SearchHistoryItemProps {
+  handleHistoryItem: (place: string) => void;
+  handleDeleteHistoryItem: (index: number) => void;
+  place: string;
+  i: number;
```
Comment (nkv843) [inline src/components/SearchHistoryItem.tsx:8]: bad naming, pls rename

File: src/features/NavBar/SearchHistory.tsx [hunk 13]
```diff
@@ -1,23 +1,28 @@
...
+export const SearchHistory: React.FC<SearchHistoryProps> = ({
   isInputFocused,
   searchHistory,
   setSearchHistory,
 }) => {
   const { setSearchFieldVisible, setSelectedLocation } = useContext(AppContext);
 
-  const handleDeleteHistoryItem = i => {
+  const handleDeleteHistoryItem = (i: number) => {
     const newHistory = searchHistory.filter((place, ii) => ii !== i);
```
Comment (nkv843) [inline src/features/NavBar/SearchHistory.tsx:20]: renama ii and i

File: src/features/WeatherDisplay/ForecastBox.tsx [hunk 14]
```diff
@@ -1,10 +1,9 @@
-/* eslint-disable react/prop-types */
 import { ForecastDay } from '../../components';
 import { AppContext } from '../../store';
-import { useContext } from 'react';
-import './styles.scss'; // Import the SCSS file
+import  { useContext } from 'react';
```
Comment (nkv843) [inline src/features/WeatherDisplay/ForecastBox.tsx:3]: why 2 spaces ?)

File: src/functions/getFutureWeather.ts [hunk 15]
```diff
@@ -1,16 +1,20 @@
...
-export const getFutureWeather = async location => {
-  let endpoint = null;
+export const getFutureWeather = async (location: Location): Promise<FutureWeather[]> => {
+  let endpoint: string = ''
   if (location.latitude && location.longitude) {
     endpoint = `https://api.weatherbit.io/v2.0/forecast/daily?lat=${location.latitude}&lon=${location.longitude}&key=${apikey}`;
   }
   if (location.locationName) {
     endpoint = `https://api.weatherbit.io/v2.0/forecast/daily?city=${location.locationName}&key=${apikey}`;
   }
```
Comment (nkv843) [inline src/functions/getFutureWeather.ts:15]: refactor it please using https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams

and also extract in method to use in getTodayWeather

**Review by nkv843 — APPROVED** [review]

File: src/components/index.ts (hunk 9 above)
Comment (andy-liubinski-stefanini) [reply src/components/index.ts:1]: done

File: src/functions/getFutureWeather.ts (hunk 15 above)
Comment (andy-liubinski-stefanini) [reply src/functions/getFutureWeather.ts:15]: done. Awesome!

File: src/features/WeatherDisplay/ForecastBox.tsx (hunk 14 above)
Comment (andy-liubinski-stefanini) [reply src/features/WeatherDisplay/ForecastBox.tsx:3]: +

File: src/features/NavBar/SearchHistory.tsx (hunk 13 above)
Comment (andy-liubinski-stefanini) [reply src/features/NavBar/SearchHistory.tsx:20]: +

File: src/components/SearchHistoryItem.tsx (hunk 12 above)
Comment (andy-liubinski-stefanini) [reply src/components/SearchHistoryItem.tsx:8]: +

File: src/components/ForecastDay.tsx (hunk 11 above)
Comment (andy-liubinski-stefanini) [reply src/components/ForecastDay.tsx:17]: It was more of a self training. Like "what can be done with an interface"

**Pull request merged**